### ISSUE TITLE
Regression after 3.7.5: Server crashes on a particular AQL

### ISSUE THREAD

**Issue by ysmu** [arangodb/arangodb#14051]
## My Environment

* __ArangoDB Version__:        3.7.10
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start
* __Configuration__:               Default
* __Infrastructure__:               Own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        SSD
* __Used Package__:              Official Windows installer

## Steps to reproduce

Running the follow AQL query crashes an empty ArangoDB server running on 3.7.10.

```
FOR doc IN []
    LET nested_docs = (
        FOR nested_doc in []
            FILTER false
            RETURN {}
    )

    FOR name_ IN nested_docs[*].name
        COLLECT name = name_
        RETURN null
```

I encountered this problem following the upgrade from 3.7.5 to 3.7.10. I don't know which release introduced this issue, but the issue does not reproduce in 3.7.5. Nothing interesting was printed before the crash, even in `TRACE` mode. For completeness, here are the `INFO` level logs:

```bash
cmd> arangod
2021-04-21T23:58:53Z [23900] INFO [e52b0] ArangoDB 3.7.10 [win64] 64bit, using build tags/v3.7.10-0-g1bb10a05e36, VPack 0.1.33, RocksDB 6.8.0, ICU 64.2, V8 7.9.317, OpenSSL 1.1.1j  16 Feb 2021
2021-04-21T23:58:53Z [23900] INFO [43396] {authentication} Jwt secret not specified, generating...
2021-04-21T23:58:53Z [23900] INFO [144fe] using storage engine 'rocksdb'
2021-04-21T23:58:53Z [23900] INFO [3bb7d] {cluster} Starting up with role SINGLE
2021-04-21T23:58:53Z [23900] INFO [3844e] {authentication} Authentication is turned on (system only)
2021-04-21T23:58:54Z [23900] INFO [c1b63] {arangosearch} ArangoSearch maintenance: [1..1] commit thread(s), [1..1] consolidation thread(s)
2021-04-21T23:58:54Z [23900] INFO [6ea38] using endpoint 'http+tcp://127.0.0.1:8529' for non-encrypted requests
2021-04-21T23:58:56Z [23900] INFO [cf3f4] ArangoDB (version 3.7.10 [win64]) is ready for business. Have fun!

cmd>
```

**Comment by jsteemann** [comment]
Reproduced it with v3.7.10.
I verified that the problem is already fixed in latest 3.7, so v3.7.11 will fix the issue.

For reference, here is the fix: 24139039ad94125ad9efb6937deb0630d9652707

And here is the problematic code part:
```
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  3 [+0x000000001f747d39] arangodb::aql::CostEstimate::restoreEstimatedNrItems() (+0x229)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  4 [+0x00000000210ec6fa] arangodb::aql::SubqueryEndNode::estimateCost() const (+0x11a)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  5 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  6 [+0x00000000202319c4] arangodb::aql::CalculationNode::estimateCost() const (+0x2e4)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  7 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  8 [+0x0000000020263d72] arangodb::aql::EnumerateListNode::estimateCost() const (+0x612)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame  9 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 10 [+0x00000000213b4e08] arangodb::aql::CollectNode::estimateCost() const (+0x1a8)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 11 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 12 [+0x000000001f3de39e] arangodb::aql::SortNode::estimateCost() const (+0x10e)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 13 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 14 [+0x0000000020232cf4] arangodb::aql::ReturnNode::estimateCost() const (+0x2e4)
2021-04-22T10:12:53Z [26153] INFO [308c3] {crash} frame 15 [+0x000000002022f1f8] arangodb::aql::ExecutionNode::getCost() const (+0x208)

**Comment by ysmu** [comment]
Thank you for the analysis! Looking forward to the release.
